langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiChatModelIT.java

```diff
@@ -322,6 +335,135 @@ void should_support_all_string_model_names(AzureOpenAiChatModelName modelName) {
+                requestReference.set(requestContext.request());
+                requestContext.attributes().put("id", "12345");
+            }
+
+            @Override
+            public void onResponse(ChatModelResponseContext responseContext) {
+                responseReference.set(responseContext.response());
+                assertThat(responseContext.request()).isSameAs(requestReference.get());
+                assertThat(responseContext.attributes().get("id")).isEqualTo("12345");
+            }
+
+            @Override
+            public void onError(ChatModelErrorContext errorContext) {
+                fail("onError() must not be called");
+            }
+        };
+
+        AzureOpenAiChatModelName modelName = GPT_4_O;
+        double temperature = 0.7;
+        double topP = 1.0;
+        int maxTokens = 7;
+
+        AzureOpenAiChatModel model = AzureOpenAiChatModel.builder()
+            .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
+            .apiKey(System.getenv("AZURE_OPENAI_KEY"))
+            .deploymentName(deploymentName)
+            .tokenizer(new AzureOpenAiTokenizer(gptVersion))
+            .topP(topP)
+            .temperature(temperature)
+            .maxTokens(maxTokens)
+            .logRequestsAndResponses(true)
+            .listeners(singletonList(listener))
+            .build();
+
+        UserMessage userMessage = UserMessage.from("hello");
+
+        ToolSpecification toolSpecification = ToolSpecification.builder()
+            .name("add")
+            .addParameter("a", INTEGER)
+            .addParameter("b", INTEGER)
+            .build();
+
+        // when
+        AiMessage aiMessage = model.generate(singletonList(userMessage), singletonList(toolSpecification)).content();
+
+        // then
+        ChatModelRequest request = requestReference.get();
+        assertThat(request.model()).isEqualTo(modelName.toString());
+        assertThat(request.temperature()).isEqualTo(temperature);
+        assertThat(request.topP()).isEqualTo(topP);
+        assertThat(request.maxTokens()).isEqualTo(maxTokens);
+        assertThat(request.messages()).containsExactly(userMessage);
+        assertThat(request.toolSpecifications()).containsExactly(toolSpecification);
+
+        ChatModelResponse response = responseReference.get();
+        assertThat(response.id()).isNotBlank();
+        assertThat(response.model()).isNotBlank();
+        assertThat(response.tokenUsage().inputTokenCount()).isGreaterThan(0);
+        assertThat(response.tokenUsage().outputTokenCount()).isGreaterThan(0);
+        assertThat(response.tokenUsage().totalTokenCount()).isGreaterThan(0);
+        assertThat(response.finishReason()).isNotNull();
+        assertThat(response.aiMessage()).isEqualTo(aiMessage);
+    }
+
+    @ParameterizedTest(name = "Deployment name {0} using {1}")
+    @CsvSource({
+        "gpt-4o,        gpt-4o"
+    })
+    void should_listen_error(String deploymentName, String gptVersion) {
```

Comment:
`deploymentName` and `gptVersion` unused

Reply:
removed unused